pkgs/development/tools/gomacro/default.nix

```
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gomacro";
+  version = "2.7";
```

Technically this package is "post 2.7" thus I would mention it somehow. In this case I would to do something like this:

```suggestion
  version = "20210131-${lib.strings.substring 0 7 rev}";
  rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
```

Using a date make it easy for a human to get if the package is up-to-date or not. Adding a part of the hash allows one to go directly to the state in the source which was used. 

There are other approaches of course, e. g. using `unstable` and a like.